pkg/terraform/provider_runner.go

```diff
@@ -0,0 +1,93 @@
+/*
+Copyright 2022 The Crossplane Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package terraform
+
+import (
+	"sync"
+
+	"github.com/crossplane/crossplane-runtime/pkg/logging"
+	"k8s.io/apimachinery/pkg/util/clock"
+	"k8s.io/utils/exec"
+)
+
+// NativeProviderRunner is the interface for running
+// Terraform native provider processes in the shared
+// gRPC server mode
+type NativeProviderRunner interface {
+	StartSharedServer() (string, error)
+}
+
+// NoOpProviderRunner is a no-op NativeProviderRunner
+type NoOpProviderRunner struct{}
```

Comment:
Having a `NewNoOpProviderRunner() NoOpProviderRunner` would make it more consistent I think.